app/forms/assets/CurrentValueFormProvider.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.assets
+
+import forms.Validation
+import forms.mappings.Mappings
+import javax.inject.Inject
+import play.api.data.Form
+
+class CurrentValueFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value"-> currency("currentValue.error.required")
+      .verifying(
+        firstError(
+          isNotEmpty("value", "currentValue.error.required"),
+          maxLength(12, "currentValue.error.length"),
+          regexp(Validation.onlyNumbersRegex, "currentValue.error.invalidFormat"),
+          regexp(Validation.decimalCheck, "currentValue.error.wholeNumber")
+        )
+      )
+    )
+}
```

Comment:
Also needs `minimumValue("1", "currentValue.error.zero")` despite the schema saying 0 is a valid answer - this is incorrect